### PR TITLE
Remove caching from Image, Object, and Workspace

### PR DESCRIPTION
**Pull request by mcquin**
Resolves https://github.com/CellProfiler/CellProfiler/issues/2205 and:

* removes method `cellprofiler.workspace.cache`
* removes HDF5 imports and caching from `cellprofiler.object`
* removes unused class `cellprofiler.measurement.ImageSetCache`
* optimizes imports in `cellprofiler.object`

### CODE REVIEW

File: cellprofiler/object.py [hunk 1]
```diff
@@ -1,13 +1,11 @@
 """ CellProfiler.Objects.py - represents a labelling of objects in an image
 """
 
-import numpy as np
-from centrosome.cpmorphology import all_connected_components
-from centrosome.index import Indexes, all_pairs
-from centrosome.outline import outline
-from scipy.sparse import coo_matrix
-
+import centrosome.index
```
Comment (0x00B1) [inline cellprofiler/object.py]: Unrelated?

File: cellprofiler/object.py [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
+import centrosome.outline
```
Comment (0x00B1) [inline cellprofiler/object.py]: Unrelated?

File: cellprofiler/object.py [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 import decorator
+import numpy
```
Comment (0x00B1) [inline cellprofiler/object.py]: Unrelated?

File: cellprofiler/object.py [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
+import scipy.sparse
```
Comment (0x00B1) [inline cellprofiler/object.py]: Unrelated?

File: cellprofiler/object.py [hunk 5]
```diff
@@ -70,7 +68,7 @@ def get_segmented(self):
             "Operation failed because objects were not initialized"
         dense, indices = self.__segmented.get_dense()
         assert len(dense) == 1, "Operation failed because objects overlapped. Please try with non-overlapping objects"
-        assert np.all(np.array(dense.shape[1:-2]) == 1), \
+        assert numpy.all(numpy.array(dense.shape[1:-2]) == 1), \
```
Comment (0x00B1) [inline cellprofiler/object.py]: Unrelated?

File: cellprofiler/object.py [hunk 6]
```diff
@@ -107,12 +104,11 @@ def get_ijv(self):
         The ijv format is a list of i,j coordinates in slots 0 and 1
         and the label at the pixel in slot 2.
         '''
-        from cellprofiler.utilities.hdf5_dict import HDF5ObjectSet
         sparse = self.__segmented.get_sparse()
-        return np.column_stack(
+        return numpy.column_stack(
```
Comment (0x00B1) [inline cellprofiler/object.py]: Unrelated?

File: cellprofiler/object.py [hunk 7]
```diff
@@ -246,37 +228,37 @@ def make_ijv_outlines(self, colors):
         # the most similar colors in the color space for objects that
         # don't overlap.
         #
-        all_labels = [(outline(label), indexes) for label, indexes in self.get_labels()]
-        image = np.zeros(list(all_labels[0][0].shape) + [3], np.float32)
+        all_labels = [(centrosome.outline.outline(label), indexes) for label, indexes in self.get_labels()]
```
Comment (0x00B1) [inline cellprofiler/object.py]: Unrelated?

File: cellprofiler/measurement.py [hunk 8]
```diff
@@ -1664,14 +1663,6 @@ def get_image_provider(self, name):
         assert len(providers) == 1, "More than one provider of the %s image" % name
         return providers[0]
 
-    def remove_image_provider(self, name):
```
Comment (0x00B1) [inline cellprofiler/measurement.py]: Unrelated?

File: cellprofiler/pipeline.py [hunk 9]
```diff
@@ -2105,14 +2105,7 @@ def run_module(self, module, workspace):
...
         '''
-        try:
-            module.run(workspace)
-        finally:
-            try:
-                workspace.cache()
-            except:
-                logger.error("Exception while trying to cache",
-                             exc_info=True)
+        module.run(workspace)
```
Comment (0x00B1) [inline cellprofiler/pipeline.py]: smdh

**Comment by 0x00B1** [comment]
How would you feel about benchmarking your pull request? It could be simple, e.g. time CellProfiler with `time` for ten, fifty, and one hundred images.

**Comment by mcquin** [comment]
Benchmarks:

|Pipeline|Num Images|With Caching (master)|Without caching (PR #2229)|
|---|---|---|---|
|Tumors|2|real: 0m4.712s<br>user: 0m4.454s<br>sys: 0m1.102s |real: 0m4.514s<br>user: 0m4.457s<br>sys: 0m0.906s|
|Illumination Correction<br>(Example 1, All Methods)|72|real: 1m19.995s<br>user: 1m30.703s<br>sys: 0m2.675s|real: 1m16.933s<br>user: 1m31.553s<br>sys: 0m2.241s|
